contracts/interchain-token-service/tests/executable.rs

```diff
@@ -0,0 +1,203 @@
+mod utils;
+
+use axelar_gateway::testutils::{generate_proof, get_approve_hash};
+use axelar_gateway::types::Message as GatewayMessage;
+use axelar_soroban_std::traits::BytesExt;
+use axelar_soroban_std::{assert_invoke_auth_err, events};
+use interchain_token_service::types::{HubMessage, InterchainTransfer, Message};
+use soroban_sdk::token;
+use soroban_sdk::xdr::ToXdr;
+use soroban_sdk::{testutils::Address as _, vec, Address, Bytes, BytesN, String};
+use utils::{register_chains, setup_env, setup_its_token, HUB_CHAIN};
+
+mod test {
+    use axelar_soroban_std::{events::Event, impl_event_testutils};
+    use core::fmt::Debug;
+    use interchain_token_service::executable::InterchainTokenExecutableInterface;
+    use soroban_sdk::{
+        contract, contractimpl, contracttype, Address, Bytes, BytesN, Env, IntoVal, String, Symbol,
+        Topics, Val,
+    };
+
+    #[contract]
+    pub struct ExecutableContract;
+
+    #[contracttype]
+    #[derive(Clone, Debug)]
+    pub enum DataKey {
+        InterchainTokenService,
+        Message,
+    }
+
+    #[derive(Debug, PartialEq, Eq)]
+    pub struct ExecutedEvent {
+        pub source_chain: String,
+        pub message_id: String,
+        pub source_address: Bytes,
+        pub payload: Bytes,
+        pub token_id: BytesN<32>,
+        pub token_address: Address,
+        pub amount: i128,
+    }
```

Comment:
Since the `payload` will be in the data section and can be very large, would it be better to have the payload as the last parameter for the ExecutedEvent?

```suggestion
    #[derive(Debug, PartialEq, Eq)]
    pub struct ExecutedEvent {
        pub source_chain: String,
        pub message_id: String,
        pub source_address: Bytes,
        pub token_id: BytesN<32>,
        pub token_address: Address,
        pub amount: i128,
        pub payload: Bytes,
    }
```